proxy_luafgen.c

```diff
@@ -323,52 +394,16 @@ static void mcp_funcgen_cleanup(lua_State *L, mcp_funcgen_t *fgen) {
     }
 
     // decrement the slot counter
-    const char *name = NULL;
-    if (fgen->name_ref) {
-        lua_rawgeti(L, LUA_REGISTRYINDEX, fgen->name_ref);
-        name = lua_tostring(L, -1);
-    } else if (fgen->router.type != FGEN_ROUTER_NONE) {
-        name = "mcp_router";
-    } else {
-        name = "anonymous";
-    }
     LIBEVENT_THREAD *t = PROXY_GET_THR(L);
-    mcp_sharedvm_delta(t->proxy_ctx, SHAREDVM_FGEN_IDX, name, -1);
+    mcp_sharedvm_delta(t->proxy_ctx, SHAREDVM_FGEN_IDX, fgen->name, -1);
 
     // Walk every request context and issue cleanup.
     for (int x = 0; x < fgen->total; x++) {
```

Comment:
The actual list length is `free_max`. Prior code would only free when `free == total`.

Need to change this to one of:
- walk the local registry (for each non-integer value in table)
- walk until `free_max`, but during alloc/realloc of the free list we need to set all new entries to NULL